### PR TITLE
re-enable helm

### PR DESCRIPTION
**Pull request by rkuo-danswer**
## Description
Fixes DAN-953.
Fixes DAN-954.
Fixes DAN-563.

## How Has This Been Tested?
[Describe the tests you ran to verify your changes]

## Accepted Risk (provide if relevant)
N/A

## Related Issue(s) (provide if relevant)
N/A

## Mental Checklist:
- All of the automated tests pass
- All PR comments are addressed and marked resolved
- If there are migrations, they have been rebased to latest main
- If there are new dependencies, they are added to the requirements
- If there are new environment variables, they are added to all of the deployment methods
- If there are new APIs that don't require auth, they are added to PUBLIC_ENDPOINT_SPECS
- Docker images build and basic functionalities work
- Author has done a final read through of the PR right before merge

## Backporting (check the box to trigger backport action)
Note: You have to check that the action passes, otherwise resolve the conflicts manually and tag the patches.
- [ ] This PR should be backported (make sure to check that the backport attempt succeeds)

### CODE REVIEW

**Review by yuhongsun96 — APPROVED** [review]
What are the ways we tested this? Does this work locally on minikube and on EKS?

File: deployment/helm/charts/danswer/templates/configmap.yaml [hunk 1]
```diff
@@ -7,7 +7,7 @@ metadata:
 data:
   INTERNAL_URL: "http://{{ include "danswer-stack.fullname" . }}-api-service:{{ .Values.api.service.port | default 8080 }}"
   POSTGRES_HOST: {{ .Release.Name }}-postgresql
-  VESPA_HOST: "document-index-service"
+  VESPA_HOST: da-vespa-0.vespa-service
```
Comment (yuhongsun96) [inline deployment/helm/charts/danswer/templates/configmap.yaml:10]: Why is this changed? Does this have to do with the vespa charts we're hosting?

**Review by yuhongsun96 — APPROVED** [review]
Oh also, one other quick question. Since Vespa is a container and we've done it as a chart, why do we not do it for our backend container and web container?

File: deployment/helm/charts/danswer/templates/configmap.yaml (hunk 1 above)
Comment (rkuo-danswer) [reply deployment/helm/charts/danswer/templates/configmap.yaml:10]: i'm not sure the naming scheme used before ever actually worked.  This does work, but I'd like to tweak it further at some point.

**Comment by rkuo-danswer** [comment]
@yuhongsun96 

> What are the ways we tested this? Does this work locally on minikube and on EKS?
Just tried it on EKS and it does work (although EKS has its own specific nits to work through ... but none requiring helm chart changes.). Works on kind as shown in the helm chart testing and minikube last I checked.

> Oh also, one other quick question. Since Vespa is a container and we've done it as a chart, why do we not do it for our backend container and web container?

It's a thought and might help more advanced deployments with deploying different parts of the app.
